java/hiro-client/src/main/java/co/arago/hiro/client/rest/DefaultWebSocketClient.java

```diff
@@ -20,124 +21,227 @@
 import org.asynchttpclient.ws.WebSocketListener;
 import org.asynchttpclient.ws.WebSocketUpgradeHandler;
 
-public class DefaultWebSocketClient implements WebSocketClient {
+public final class DefaultWebSocketClient implements WebSocketClient {
 
+  private static final int MAX_RETRIES = 20;
```

Comment:
why do we need to retry 20 x 1000 ms? for which case is it? 
if it fails to send 3-5 times then it will probably not recover
just slowing down error response?

Reply:
adjusted